// Validating benchmarks:
// ***** BenchmarkRunner: Start   *****
// ***** Found 5 benchmark(s) in total *****
// ***** Building 1 exe(s) in Parallel: Start   *****
// start dotnet restore  /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3
// command took 1.52s and exited with 0
// start dotnet build -c Release  --no-restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3
// command took 2.58s and exited with 0
// ***** Done, took 00:00:04 (4.33 sec)   *****
// Found 5 benchmarks:
//   ReduceBenchmark.LinqAggregate: DefaultJob
//   ReduceBenchmark.PLinqAggregate: DefaultJob
//   ReduceBenchmark.ForLoopAggregate: DefaultJob
//   ReduceBenchmark.ForEachLoopAggregate: DefaultJob
//   ReduceBenchmark.ParallelForLoop: DefaultJob

Setup power plan (GUID: 8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c FriendlyName: High performance)// **************************
// Benchmark: ReduceBenchmark.LinqAggregate: DefaultJob
// *** Execute ***
// Launch: 1 / 1
// Execute: dotnet "12965301-cd74-4dbd-9a74-ea1d226cecf3.dll" --benchmarkName "LinqBenchmarks.ReduceBenchmark.LinqAggregate" --job "Default" --benchmarkId 0 in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\bin\Release\netcoreapp3.0
// BeforeAnythingElse

System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at LinqBenchmarks.ReduceBenchmark..ctor() in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\Program.cs:line 19
   at BenchmarkDotNet.Autogenerated.Runnable_0..ctor() in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 181
   at BenchmarkDotNet.Autogenerated.Runnable_0.Run(IHost host, String benchmarkName) in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 128
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(String[] args) in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 52
// AfterAll
ExitCode != 0
No more Benchmark runs will be launched as NO measurements were obtained from the previous run!


// **************************
// Benchmark: ReduceBenchmark.PLinqAggregate: DefaultJob
// *** Execute ***
// Launch: 1 / 1
// Execute: dotnet "12965301-cd74-4dbd-9a74-ea1d226cecf3.dll" --benchmarkName "LinqBenchmarks.ReduceBenchmark.PLinqAggregate" --job "Default" --benchmarkId 1 in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\bin\Release\netcoreapp3.0
// BeforeAnythingElse

System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at LinqBenchmarks.ReduceBenchmark..ctor() in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\Program.cs:line 19
   at BenchmarkDotNet.Autogenerated.Runnable_1..ctor() in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 825
   at BenchmarkDotNet.Autogenerated.Runnable_1.Run(IHost host, String benchmarkName) in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 772
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(String[] args) in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 52
// AfterAll
ExitCode != 0
No more Benchmark runs will be launched as NO measurements were obtained from the previous run!


// **************************
// Benchmark: ReduceBenchmark.ForLoopAggregate: DefaultJob
// *** Execute ***
// Launch: 1 / 1
// Execute: dotnet "12965301-cd74-4dbd-9a74-ea1d226cecf3.dll" --benchmarkName "LinqBenchmarks.ReduceBenchmark.ForLoopAggregate" --job "Default" --benchmarkId 2 in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\bin\Release\netcoreapp3.0
// BeforeAnythingElse

System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at LinqBenchmarks.ReduceBenchmark..ctor() in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\Program.cs:line 19
   at BenchmarkDotNet.Autogenerated.Runnable_2..ctor() in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 1469
   at BenchmarkDotNet.Autogenerated.Runnable_2.Run(IHost host, String benchmarkName) in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 1416
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(String[] args) in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 52
// AfterAll
ExitCode != 0
No more Benchmark runs will be launched as NO measurements were obtained from the previous run!


// **************************
// Benchmark: ReduceBenchmark.ForEachLoopAggregate: DefaultJob
// *** Execute ***
// Launch: 1 / 1
// Execute: dotnet "12965301-cd74-4dbd-9a74-ea1d226cecf3.dll" --benchmarkName "LinqBenchmarks.ReduceBenchmark.ForEachLoopAggregate" --job "Default" --benchmarkId 3 in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\bin\Release\netcoreapp3.0
// BeforeAnythingElse

System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at LinqBenchmarks.ReduceBenchmark..ctor() in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\Program.cs:line 19
   at BenchmarkDotNet.Autogenerated.Runnable_3..ctor() in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 2113
   at BenchmarkDotNet.Autogenerated.Runnable_3.Run(IHost host, String benchmarkName) in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 2060
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(String[] args) in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 52
// AfterAll
ExitCode != 0
No more Benchmark runs will be launched as NO measurements were obtained from the previous run!


// **************************
// Benchmark: ReduceBenchmark.ParallelForLoop: DefaultJob
// *** Execute ***
// Launch: 1 / 1
// Execute: dotnet "12965301-cd74-4dbd-9a74-ea1d226cecf3.dll" --benchmarkName "LinqBenchmarks.ReduceBenchmark.ParallelForLoop" --job "Default" --benchmarkId 4 in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\bin\Release\netcoreapp3.0
// BeforeAnythingElse

System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at LinqBenchmarks.ReduceBenchmark..ctor() in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\Program.cs:line 19
   at BenchmarkDotNet.Autogenerated.Runnable_4..ctor() in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 2757
   at BenchmarkDotNet.Autogenerated.Runnable_4.Run(IHost host, String benchmarkName) in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 2704
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(String[] args) in F:\Eduardo\Documents\Code\C#\LambdaBenchmarks\LinqBenchmarks\bin\Release\netcoreapp3.0\12965301-cd74-4dbd-9a74-ea1d226cecf3\12965301-cd74-4dbd-9a74-ea1d226cecf3.notcs:line 52
// AfterAll
ExitCode != 0
No more Benchmark runs will be launched as NO measurements were obtained from the previous run!


Successfully reverted power plan (GUID: 8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c FriendlyName: High performance)
// ***** BenchmarkRunner: Finish  *****

// * Export *
  BenchmarkDotNet.Artifacts\results\LinqBenchmarks.ReduceBenchmark-report.csv
  BenchmarkDotNet.Artifacts\results\LinqBenchmarks.ReduceBenchmark-report-github.md
  BenchmarkDotNet.Artifacts\results\LinqBenchmarks.ReduceBenchmark-report.html

// * Detailed results *
ReduceBenchmark.LinqAggregate: DefaultJob
Runtime = ; GC = 
There are not any results runs

ReduceBenchmark.PLinqAggregate: DefaultJob
Runtime = ; GC = 
There are not any results runs

ReduceBenchmark.ForLoopAggregate: DefaultJob
Runtime = ; GC = 
There are not any results runs

ReduceBenchmark.ForEachLoopAggregate: DefaultJob
Runtime = ; GC = 
There are not any results runs

ReduceBenchmark.ParallelForLoop: DefaultJob
Runtime = ; GC = 
There are not any results runs

// * Summary *

BenchmarkDotNet=v0.11.5, OS=Windows 10.0.18362
AMD Ryzen 5 1600, 1 CPU, 12 logical and 6 physical cores
.NET Core SDK=3.0.100
  [Host] : .NET Core 3.0.0 (CoreCLR 4.700.19.46205, CoreFX 4.700.19.46214), 64bit RyuJIT  [AttachedDebugger]


|               Method | Mean | Error |
|--------------------- |-----:|------:|
|        LinqAggregate |   NA |    NA |
|       PLinqAggregate |   NA |    NA |
|     ForLoopAggregate |   NA |    NA |
| ForEachLoopAggregate |   NA |    NA |
|      ParallelForLoop |   NA |    NA |

Benchmarks with issues:
  ReduceBenchmark.LinqAggregate: DefaultJob
  ReduceBenchmark.PLinqAggregate: DefaultJob
  ReduceBenchmark.ForLoopAggregate: DefaultJob
  ReduceBenchmark.ForEachLoopAggregate: DefaultJob
  ReduceBenchmark.ParallelForLoop: DefaultJob

// * Warnings *
Environment
  Summary -> Benchmark was executed with attached debugger

// * Legends *
  Mean  : Arithmetic mean of all measurements
  Error : Half of 99.9% confidence interval
  1 ns  : 1 Nanosecond (0.000000001 sec)

// ***** BenchmarkRunner: End *****
// ** Remained 0 benchmark(s) to run **
Run time: 00:00:01 (1.42 sec), executed benchmarks: 5

Global total time: 00:00:05 (5.76 sec), executed benchmarks: 5
// * Artifacts cleanup *
